xmake/core/project/project.lua

```diff
@@ -897,9 +897,11 @@ function project.ordertargets()
     local ordertargets = project._memcache():get("ordertargets")
     if not ordertargets then
         local targets = project.targets()
+        local sorted_targets = table.values(targets)
+        table.sort(sorted_targets, function (a,b) return a:name() < b:name() end)
         ordertargets = {}
         local targetrefs = {}
-        for _, t in pairs(targets) do
+        for _, t in ipairs(sorted_targets) do
             instance_deps.sort_deps(targets, ordertargets, targetrefs, t)
```

Comment:
t:get("deps") is not fixed order, but it is just an issue your project.

https://github.com/NazaraEngine/NazaraEngine/blob/0a8048809c014bf41fa4c50f0cff4fc689832635/xmake.lua#L117

```lua
for name, module in pairs(rendererBackends) do
	ModuleTarget(name, module)
end
```

you should use `table.orderpairs`

Reply:
you're right, however I'm testing project generation on this project: https://github.com/NazaraEngine/ShaderLang (which should be stable)